sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -269,7 +269,10 @@ object GpuDivModLike {
 }
 
 trait GpuDivModLike extends CudfBinaryArithmetic {
-  lazy val failOnError: Boolean = ShimLoader.getSparkShims.shouldFailDivByZero()
+  val failOnErrorOverride: Option[Boolean] = None
+
```

Comment:
let us try without `failOnErrorOverride`, just make failOnError non-lazy, so we can override it

Reply:
You can override a lazy val

```
override lazy val failOnError: Boolean = failOnErrorOverride.getOrElse(GpuDivModeLike.failOnError)
```

But I am fine with keeping this as is.

Reply:
yes, I just meant you can't use lazy as a parameter.